src/main/java/core/basesyntax/dataservice/DataConverterImpl.java

```
@@ -0,0 +1,28 @@
+package core.basesyntax.dataservice;
+
+import core.basesyntax.transactions.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int OPERATION = 0;
+    private static final int FRUIT_TYPE = 1;
+    private static final int NUMBER = 2;
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        data.stream()
+                .map(array -> array.split(SEPARATOR))
+                .forEach(strings -> {
+                    FruitTransaction fruitTransaction = new FruitTransaction();
+                    fruitTransaction.setFruit(strings[FRUIT_TYPE]);
+                    fruitTransaction.setQuantity(Integer.parseInt(strings[NUMBER]));
```

What happens if quantity in not a number? Also we want to check if it's not negative before processing any transaction. It's called **fail fast** principle. It's better to use `for loop` and wrap this line in `try catch` block to throw an `IllegalArgumentException`